R/construct_design.R

```diff
@@ -62,60 +56,110 @@
 #'
 #' summary(diagnosis)
 #' }
-#'
-declare_design <- function(...) {
+#' 
+#' @importFrom rlang enexpr expr_deparse
+#' 
+#' @export
+`+.dd` <- function(lhs, rhs){
+  
+  # two cases
+  # 1. lhs is a step
+  # 2. lhs is a design
+  
+  browser()
+  
+  if (!inherits(rhs, "dd")) {
+    stop("The right hand side does not appear to be a dd object. Can you wrap the step with `tidy_step()`?", call. = FALSE)
+  }
+  
+  if (inherits(lhs, "design")) {
+    
+    # browser()
+    auto_steps <- sapply(lhs, function(x) { lbl <- attr(x, "label"); !is.null(lbl) && grepl("Autogenerated by ", lbl) })
+    if (any(auto_steps)) {
+      lhs <- lhs[!auto_steps]
+    }
+    
+    ret <- append(lhs, rhs)
+    
+    # browser()
+    
+    names(ret)[length(ret)] <- expr_text(enexpr(rhs), nlines = 1)
+    
+    ret <- do.call(construct_design, ret)
+    
+  } else {
+    
+    ret <- append(lhs, rhs) 
+    
+    names(ret) <- c(expr_text(enexpr(lhs), nlines = 1), expr_text(enexpr(rhs), nlines = 1))
+    
+    ret <- do.call(construct_design, ret)
```

Comment:
This part looks copy-pasted, needs another pass.

Reply:
also use eval_tidy here instead of do.call, otherwise you are forcing the evaluation of rhs too early.

Reply:
thanks, yes, we are working on a tidy version. (it isn't copy-pasted.)